File: guides/micronaut-dependency-injection-types/kotlin/src/main/kotlin/example/micronaut/constructor/MessageController.kt
package example.micronaut.constructor

import example.micronaut.MessageService
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces

@Controller("/constructor")
class MessageController(val messageService: MessageService) {
    @Get
    @Produces(MediaType.TEXT_PLAIN)
    fun index() = messageService.compose()
}

File: guides/micronaut-dependency-injection-types/python/src/example/micronaut/constructor/message_controller.py
from micronaut.http import MediaType
from micronaut.http.annotation import Controller, Get

from example.micronaut.message_service import MessageService


@Controller("/constructor")
class MessageController:
    def __init__(self, message_service: MessageService):
        self.message_service = message_service

    @Get(produces=MediaType.TEXT_PLAIN)
    def index(self) -> str:
        return self.message_service.compose()
